### PR TITLE
[BUGFIX] PromQL: Adds test for `absent`, `absent_over_time` and `deriv` func with histograms

### PR DESCRIPTION
**Pull request by NeerajGartia21**
Related to #13934 

This PR adds tests for `absent`, `absent_over_time` and `deriv` functions with native histograms to verify that they works perfectly with histograms.

CC: @beorn7

### CODE REVIEW

File: promql/promqltest/testdata/functions.test [hunk 1]
```diff
@@ -244,6 +244,8 @@ clear
 load 5m
 	testcounter_reset_middle_total	0+10x4 0+10x5
 	http_requests_total{job="app-server", instance="1", group="canary"}		0+80x10
+	testcounter_reset_middle_mix	0+10x4 0+10x5 {{schema:0 sum:1 count:1}} {{schema:1 sum:2 count:2}}
+	http_requests_mix{job="app-server", instance="1", group="canary"}		0+80x10 {{schema:0 sum:1 count:1}}
```
Comment (beorn7) [inline promql/promqltest/testdata/functions.test:248]: Let's also test a pur histogram series (which should result in an empty vector).

File: promql/promqltest/testdata/functions.test [hunk 2]
```diff
@@ -256,6 +258,13 @@ eval instant at 50m deriv(http_requests_total{group="canary", instance="1", job=
 eval instant at 50m deriv(testcounter_reset_middle_total[100m])
 	{} 0.010606060606060607
 
+# deriv should ignore histograms.
+eval instant at 110m deriv(http_requests_mix{group="canary", instance="1", job="app-server"}[110m])
+	{group="canary", instance="1", job="app-server"} 0.26666666666666666
+
+eval instant at 100m deriv(testcounter_reset_middle_mix[110m])
+	{} 0.010606060606060607
```
Comment (beorn7) [inline promql/promqltest/testdata/functions.test:266]: I actually think these two should have an info annotation, similar to what we plan for ..._over_time (quote from #13934: "All other ..._over_time functions silently ignore histograms. Which (probably) yields the right result, but there should be an info annotation whenever a histogram is ignored, similar to what is planned for aggregators in the table above.").

You could leave the test as is, but then we need to remember to add the info annotation later. Or you turn this into eval_info right now and add the fix in the same PR.

BTW: With a pure histogram series, ignoring of the histograms should be without annotation (that's why I think we should also have a test for the pure histogram case).

**Review by beorn7 — CHANGES_REQUESTED** [review]

File: promql/promqltest/testdata/functions.test (hunk 2 above)
Comment (NeerajGartia21) [reply promql/promqltest/testdata/functions.test:266]: Yeah `deriv` should have an info annotation. 
I think it's better to leave is as it is and I'll add another later in another PR while fixing `double_exponential_smoothing()` and `predict_linear()`.

File: promql/promqltest/testdata/functions.test (hunk 1 above)
Comment (NeerajGartia21) [reply promql/promqltest/testdata/functions.test:248]: Done!

**Review by beorn7 — APPROVED** [review]
Thank you.

**Pull request merged**